File: server/app/models/travel.py
from sqlalchemy import Column, Integer, String, Float, Date, DateTime, Boolean, ForeignKey, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database import Base


class Trip(Base):
    __tablename__ = "trips"
    id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String, nullable=False)
    destination = Column(String, nullable=False)
    country = Column(String, default="")
    start_date = Column(Date, nullable=True)
    end_date = Column(Date, nullable=True)
    budget = Column(Float, default=0.0)
    currency = Column(String, default="USD")
    status = Column(String, default="planning")  # planning | active | completed | cancelled
    notes = Column(Text, default="")
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    itinerary = relationship(
        "ItineraryItem", back_populates="trip",
        cascade="all, delete-orphan",
        order_by="ItineraryItem.day_offset, ItineraryItem.id",
    )
    packing_items = relationship(
        "PackingItem", back_populates="trip",
        cascade="all, delete-orphan",
        order_by="PackingItem.category, PackingItem.id",
    )


class ItineraryItem(Base):
    __tablename__ = "itinerary_items"
    id = Column(Integer, primary_key=True, autoincrement=True)
    trip_id = Column(Integer, ForeignKey("trips.id"), nullable=False)
    day_offset = Column(Integer, default=0)  # 0 = day 1
    time_label = Column(String, default="")  # e.g. "9:00 AM"
    title = Column(String, nullable=False)
    description = Column(Text, default="")
    location = Column(String, default="")
    estimated_cost = Column(Float, default=0.0)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    trip = relationship("Trip", back_populates="itinerary")


class PackingItem(Base):
    __tablename__ = "packing_items"
    id = Column(Integer, primary_key=True, autoincrement=True)
    trip_id = Column(Integer, ForeignKey("trips.id"), nullable=False)
    name = Column(String, nullable=False)
    category = Column(String, default="General")
    packed = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    trip = relationship("Trip", back_populates="packing_items")

File: apps/travel-planner/src/types/travel.ts
export type TripStatus = 'planning' | 'active' | 'completed' | 'cancelled'

export interface Trip {
  id: number
  title: string
  destination: string
  country: string
  start_date: string | null
  end_date: string | null
  budget: number
  currency: string
  status: TripStatus
  notes: string
  created_at: string
}

export interface TripWithDetails extends Trip {
  itinerary: ItineraryItem[]
  packing_items: PackingItem[]
}

export interface ItineraryItem {
  id: number
  trip_id: number
  day_offset: number
  time_label: string
  title: string
  description: string
  location: string
  estimated_cost: number
  notes: string
  created_at: string
}

export interface PackingItem {
  id: number
  trip_id: number
  name: string
  category: string
  packed: boolean
  created_at: string
}
